Improved py36 support

eserge opened the pull request:
The changes made here are dealing with the differences between iteration functions in py3 and py2. Specifically, `dnsq` returns a result of a filter function from its `mx_hosts_for` function. In Python 2 it's a list, but for py3 it's an iterator. Later flanker tries to get a `len` of that list which is impossible with an iterator. So this pull request deals with this by converting iterator to a list. 

Other than that there are some improvements in tests.

eserge commented:
Yes, sure. If this goes well, I'm planning to make couple more commits

horkhe commented on tests/addresslib/validator_test.py:
```diff
@@ -11,6 +12,9 @@
 from flanker.addresslib import address, validate
 
 
+PY3 = sys.version_info[0] == 3
```
Everywhere else we use `six.PY3` instead, please do the same for consistency.

eserge commented:
Made the change about six.PY3 that you requested. Also reverted expanded if...else back to ternary. It seems to me that the part of the condition where there is a dot at the end is not reachable by the execution, because the domain comes to this function after being processed by the email validation and it doesn't seem to allow domains in FQDN form with dot at the end. 

Also reverted version change in setup.py as to not to impose any particular version for this change to be released in.

horkhe commented:
Could you please squash your commits, and then I will merge.

horkhe commented:
@eserge thank you for your contribution.
